staging/src/k8s.io/kubectl/pkg/cmd/proxy/proxy.go

```diff
@@ -70,96 +93,111 @@ var (
 		kubectl proxy --api-prefix=/k8s-api`))
 )
 
+// NewProxyOptions creates the options for proxy
+func NewProxyOptions(ioStreams genericclioptions.IOStreams) *ProxyOptions {
+	return &ProxyOptions{
+		IOStreams: ioStreams,
+	}
+}
+
 // NewCmdProxy returns the proxy Cobra command
-func NewCmdProxy(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+func NewCmdProxy(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	o := NewProxyOptions(ioStreams)
+
 	cmd := &cobra.Command{
 		Use:                   "proxy [--port=PORT] [--www=static-dir] [--www-prefix=prefix] [--api-prefix=prefix]",
 		DisableFlagsInUseLine: true,
 		Short:                 i18n.T("Run a proxy to the Kubernetes API server"),
 		Long:                  proxyLong,
 		Example:               proxyExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			err := RunProxy(f, streams.Out, cmd)
-			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Complete(f))
```

Comment:
The order is wrong, first you complete the `ProxyOptions` struct, then you validate it and finally run the command.